File: app/src/main/java/com/xyoye/dandanplay/utils/net/service/ResRetrofitService.java
package com.xyoye.dandanplay.utils.net.service;

import com.xyoye.dandanplay.bean.AnimeTypeBean;
import com.xyoye.dandanplay.bean.MagnetBean;
import com.xyoye.dandanplay.bean.SubGroupBean;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by xyoye on 2020/2/23.
 */

public interface ResRetrofitService {

    @GET("/type")
    Observable<AnimeTypeBean> getAnimeType();

    @GET("/list")
    Observable<MagnetBean> searchMagnet(@Query("keyword") String keyword, @Query("type") String typeId, @Query("subgroup") String subGroupId);

    @GET("/subgroup")
    Observable<SubGroupBean> getSubGroup();
}

File: common_component/src/main/java/com/xyoye/common_component/network/service/ResRetrofitService.kt
package com.xyoye.common_component.network.service

import com.xyoye.data_component.data.MagnetResourceData
import com.xyoye.data_component.data.MagnetSubgroupData
import com.xyoye.data_component.data.MagnetTypeData
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by xyoye on 2020/8/19.
 */

interface ResRetrofitService {

    @GET("/list")
    suspend fun searchMagnet(
        @Query("keyword") keyword: String,
        @Query("type") typeId: String?,
        @Query("subgroup") subgroupId: String?
    ): MagnetResourceData

    @GET("/type")
    suspend fun getMagnetType(): MagnetTypeData

    @GET("/subgroup")
    suspend fun getMagnetSubgroup(): MagnetSubgroupData
}